File: java/src/main/java/com/edulink/gui/models/assistance/HelpRequest.java
package com.edulink.gui.models.assistance;

import java.sql.Timestamp;

public class HelpRequest {
    private int id;
    private String title;
    private String description;
    private String status;
    private int bounty;
    private boolean isTicket;
    private Timestamp createdAt;
    private Integer studentId;
    private Integer sessionId;
    private String category;
    private String difficulty;
    private String closeReason;
    private Integer tutorId;          // tutor who accepted this request
    private String jitsiRoomId;       // live video room tied to the session
    private String sessionSummary;    // AI-generated summary after session closes

    public HelpRequest() {
        this.status = "OPEN";
        this.bounty = 0;
        this.isTicket = false;
        this.createdAt = new Timestamp(System.currentTimeMillis());
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    
    public int getBounty() { return bounty; }
    public void setBounty(int bounty) { this.bounty = bounty; }
    
    public boolean isTicket() { return isTicket; }
    public void setTicket(boolean ticket) { isTicket = ticket; }
    
    public Timestamp getCreatedAt() { return createdAt; }
    public void setCreatedAt(Timestamp createdAt) { this.createdAt = createdAt; }
    
    public Integer getStudentId() { return studentId; }
    public void setStudentId(Integer studentId) { this.studentId = studentId; }
    
    public Integer getSessionId() { return sessionId; }
    public void setSessionId(Integer sessionId) { this.sessionId = sessionId; }
    
    public String getCategory() { return category; }
    public void setCategory(String category) { this.category = category; }
    
    public String getDifficulty() { return difficulty; }
    public void setDifficulty(String difficulty) { this.difficulty = difficulty; }
    
    public String getCloseReason() { return closeReason; }
    public void setCloseReason(String closeReason) { this.closeReason = closeReason; }

    public Integer getTutorId() { return tutorId; }
    public void setTutorId(Integer tutorId) { this.tutorId = tutorId; }

    public String getJitsiRoomId() { return jitsiRoomId; }
    public void setJitsiRoomId(String jitsiRoomId) { this.jitsiRoomId = jitsiRoomId; }

    public String getSessionSummary() { return sessionSummary; }
    public void setSessionSummary(String sessionSummary) { this.sessionSummary = sessionSummary; }
}

File: ai/helprequest_models/classification_api/Entity/HelpRequest.php
<?php

namespace App\Entity;

use App\Repository\HelpRequestRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: HelpRequestRepository::class)]
class HelpRequest
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: 'The title cannot be empty.')]
    #[Assert\Length(
        min: 3,
        max: 255,
        minMessage: 'The title must be at least {{ limit }} characters.',
        maxMessage: 'The title cannot exceed {{ limit }} characters.'
    )]
    private ?string $title = null;

    #[ORM\Column(type: Types::TEXT)]
    #[Assert\NotBlank(message: 'The description cannot be empty.')]
    #[Assert\Length(
        min: 10,
        minMessage: 'The description must be at least {{ limit }} characters.'
    )]
    private ?string $description = null;

    #[ORM\Column(length: 20)]
    #[Assert\Choice(
        choices: ['OPEN', 'IN_PROGRESS', 'CLOSED'],
        message: 'Invalid status. Allowed: OPEN, IN_PROGRESS, CLOSED.'
    )]
    private string $status = 'OPEN';

    #[ORM\Column]
    #[Assert\NotNull(message: 'Bounty is required.')]
    #[Assert\PositiveOrZero(message: 'Bounty cannot be negative.')]
    private int $bounty = 0;

    #[ORM\Column]
    private bool $isTicket = false;

    #[ORM\Column]
    private ?\DateTimeImmutable $createdAt = null;

    #[ORM\ManyToOne(inversedBy: 'helpRequests')]
    #[ORM\JoinColumn(nullable: false)]
    private ?User $student = null;

    #[ORM\OneToOne(mappedBy: 'helpRequest', cascade: ['persist', 'remove'])]
    private ?Session $session = null;

    #[ORM\Column(length: 50, nullable: true)]
    private ?string $category = null;

    #[ORM\Column(length: 20, nullable: true)]
    private ?string $difficulty = null;

    #[ORM\Column(length: 30, nullable: true)]
    private ?string $closeReason = null;

    public function __construct()
    {
        $this->createdAt = new \DateTimeImmutable();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitle(): ?string
    {
        return $this->title;
    }

    public function setTitle(string $title): static
    {
        $this->title = $title;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;
        return $this;
    }

    public function getStatus(): string
    {
        return $this->status;
    }

    public function setStatus(string $status): static
    {
        $this->status = $status;
        return $this;
    }

    public function getBounty(): int
    {
        return $this->bounty;
    }

    public function setBounty(?int $bounty): static
    {
        $this->bounty = $bounty ?? 0;
        return $this;
    }

    public function isIsTicket(): bool
    {
        return $this->isTicket;
    }

    public function setIsTicket(bool $isTicket): static
    {
        $this->isTicket = $isTicket;
        return $this;
    }

    public function getCreatedAt(): ?\DateTimeImmutable
    {
        return $this->createdAt;
    }

    public function setCreatedAt(\DateTimeImmutable $createdAt): static
    {
        $this->createdAt = $createdAt;
        return $this;
    }

    public function getStudent(): ?User
    {
        return $this->student;
    }

    public function setStudent(?User $student): static
    {
        $this->student = $student;
        return $this;
    }

    public function getSession(): ?Session
    {
        return $this->session;
    }

    public function setSession(?Session $session): static
    {
        if ($session === null && $this->session !== null) {
            $this->session->setHelpRequest(null);
        }

        if ($session !== null && $session->getHelpRequest() !== $this) {
            $session->setHelpRequest($this);
        }

        $this->session = $session;
        return $this;
    }

    public function getCloseReason(): ?string
    {
        return $this->closeReason;
    }

    public function setCloseReason(?string $closeReason): static
    {
        $this->closeReason = $closeReason;
        return $this;
    }

    public function getCategory(): ?string
    {
        return $this->category;
    }

    public function setCategory(?string $category): static
    {
        $this->category = $category;
        return $this;
    }

    public function getDifficulty(): ?string
    {
        return $this->difficulty;
    }

    public function setDifficulty(?string $difficulty): static
    {
        $this->difficulty = $difficulty;
        return $this;
    }
}